lib/main.dart

```diff
@@ -81,3 +105,92 @@ class _MyHomePageState extends State<MyHomePage> {
     );
   }
 }
+
+// Future<Directory> getPlatformDocumentsDirectoryPathFromPlugin() {
+//   WidgetsFlutterBinding.ensureInitialized();
+//   return path_provider.getApplicationDocumentsDirectory();
+// }
+
+/// Returns Application documents directory for a platform before `WidgetsFlutterBinding.ensureInitialized` or `runApp`
+/// is used.
+Future<Directory> getPlatformDocumentsDirectoryPath(
+  String applicationPackageName,
+) async {
+  if (Platform.isAndroid) {
+    return Directory('/data/user/0/$applicationPackageName');
+  } else if (Platform.isMacOS) {
+    final applicationHomeDir = Platform.environment['HOME'];
+    return Directory('$applicationHomeDir/Documents');
+  } else if (Platform.isIOS) {
+    final bundleContainerDir = File(Platform.executable).parent;
+    // This isn't documents directory but the bundle container directory of this application.
+    // Only works on iOS simulator and will not work on a real iPhone
```

Comment:
Needs to work on a real iOS device since capturing is always done on a real device.